MdeModulePkg/Bus/Pci/PciBusDxe/PciOptionRomSupport.c

```diff
@@ -638,6 +638,65 @@ RomDecode (
   }
 }
 
+/**
+  Scan the whole Option Rom and check if the native image exists.
+
+  @param PciDevice       Pci device instance.
+
+  @retval TRUE           The native Option Rom image exists.
+  @retval FALSE          The native Option Rom image does not exist.
+
+**/
+BOOLEAN
+FindNativeOpRom (

```

Comment:
Instead of duplicating a lot of the logic in `ProcesOpRomImage()`, could you perhaps add a `BOOL NativeOnly` argument to it, and call it with `TRUE` first, and with `FALSE` only if the first call does not return anything?

Reply:
Let me think about it.
Do you mean drop this function, move the logice into `rocessorOpRomImage`, and then add an argument called `NativeOnly` to `ProcessorOpRomImage`, then call it with `TRUE` first, if not found return `EFI_NOT_FOUND`, and call it with `FALSE` again? If so, we will add some logic to the caller.